components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -949,9 +951,11 @@ private void handleLogoutResponseFromFramework(HttpServletRequest request,
             if (StringUtils.isBlank(acsUrl) && sessionDTO.getIssuer() != null) {
                 SAMLSSOServiceProviderDO serviceProviderDO = getSPConfig(SAMLSSOUtil.getTenantDomainFromThreadLocal(),
                         sessionDTO.getIssuer());
-                acsUrl = serviceProviderDO.getSloResponseURL();
-                if (StringUtils.isBlank(acsUrl)) {
-                    acsUrl = serviceProviderDO.getDefaultAssertionConsumerUrl();
+                if (serviceProviderDO != null) {
```

Comment:
Shouldnt we break the flow if SP is null?

Reply:
here acsUrl is required for error scenario paths. serviceProviderDO is obtained to set the acsUrl as a query param to saml notification.do page if it is null. This shouldn't break the flow.